src/sentry/tasks/post_process.py

```diff
@@ -116,16 +116,32 @@ def handle_owner_assignment(project, group, event):
 
 
 @instrumented_task(name="sentry.tasks.post_process.post_process_group")
-def post_process_group(event, is_new, is_regression, is_new_group_environment, **kwargs):
+def post_process_group(
+    event, is_new, is_regression, is_new_group_environment, cache_key=None, group_id=None, **kwargs
+):
     """
     Fires post processing hooks for a group.
     """
-    set_current_project(event.project_id)
-
+    from sentry.eventstore.models import Event
+    from sentry.eventstore.processing import event_processing_store
     from sentry.utils import snuba
 
     with snuba.options_override({"consistent": True}):
-        if check_event_already_post_processed(event):
+        # The event parameter will be removed after transitioning to
+        # event_processing_store is complete.
+        if cache_key and event is None:
+            data = event_processing_store.get(cache_key)
+            if not data:
+                logger.info(
+                    "post_process.skipped",
+                    extra={"cache_key": cache_key, "reason": "missing_cache"},
+                )
+                return
+            event = Event(
+                project_id=data["project"], event_id=data["event_id"], group_id=group_id, data=data
+            )
+        elif check_event_already_post_processed(event):
```

Comment:
Should this be an `if` instead of an `elif`?

Reply:
I didn't want to do a lock check when we use the cache key path. The already processed check is intended to prevent duplicate work in the situation where we lose state in the forwarding consumer and need to replay the last 15-30minutes of events again.

In the cache_key path the events would have been removed from the processing store as post_process completes. On a re-run post_process will exit early on line 135 as the cache is empty.

In the path where the event is in the MQ message we want to apply the lock check to prevent running post process twice.